Firestore/core/src/firebase/firestore/model/field_value.cc

```diff
@@ -153,6 +153,72 @@ FieldValue& FieldValue::operator=(FieldValue&& value) {
   }
 }
 
+FieldValue FieldValue::Set(FieldPath field_path, FieldValue value) const {
+  FIREBASE_ASSERT_MESSAGE(type() == Type::Object,
+                          "Cannot set field for non-object FieldValue");
+  FIREBASE_ASSERT_MESSAGE(!field_path.empty(),
+                          "Cannot set field for empty path on FieldValue");
+  const std::string& child_name = field_path.first_segment();
+  if (field_path.size() == 1) {
+    // TODO(zxu): Once immutable type is available, rewrite these.
+    ObjectValue::Map copy = object_value_.internal_value;
+    copy[child_name] = std::move(value);
+    return FieldValue::ObjectValueFromMap(std::move(copy));
+  } else {
+    ObjectValue::Map copy = object_value_.internal_value;
+    const auto iter = copy.find(child_name);
+    if (iter == copy.end() || iter->second.type() != Type::Object) {
+      copy[child_name] = FieldValue::ObjectValueFromMap({});
```

Comment:
So `Set` will create intermediate objects if necessary (i.e., it behaves like `mkdir -p`, not plain `mkdir`)? If that's correct, could you please add it to the comment in the header file?